docker.compose.image.swarm.yml

```diff
@@ -0,0 +1,108 @@
+version: "3.7"
+services:
+  db_gtm_1:
+    environment:
+      - PG_HOST=0.0.0.0
+      - PG_NODE=gtm_1
+      - PG_PORT=6666
+    image: pavouk0/postgres-xl:latest
+    command: docker-cmd-gtm
+    entrypoint: docker-entrypoint-gtm
+    volumes:
+      - db_gtm_1:/var/lib/postgresql
+    networks:
+      - db
+    #    healthcheck:
+    #      test: ["CMD", "docker-healthcheck-gtm"]
+    deploy:
+      mode: global
+  db_coord_1:
+    environment:
+      - PG_GTM_HOST=db_gtm_1
+      - PG_GTM_PORT=6666
+      - PG_HOST=0.0.0.0
+      - PG_NODE=coord_1
+      - PG_PORT=5432
+    image: pavouk0/postgres-xl:latest
+    command: docker-cmd-coord
+    entrypoint: docker-entrypoint-coord
+    volumes:
+      - db_coord_1:/var/lib/postgresql
+    depends_on:
+      - db_gtm_1
+    networks:
+      - db
+    #    healthcheck:
+    #      test: ["CMD", "docker-healthcheck-coord"]
+    deploy:
+      mode: global
+  db_coord_2:
+    environment:
+      - PG_GTM_HOST=db_gtm_1
+      - PG_GTM_PORT=6666
+      - PG_HOST=0.0.0.0
+      - PG_NODE=coord_2
+      - PG_PORT=5432
+    image: pavouk0/postgres-xl:latest
+    command: docker-cmd-coord
+    entrypoint: docker-entrypoint-coord
+    volumes:
+      - db_coord_2:/var/lib/postgresql
+    depends_on:
+      - db_gtm_1
+    networks:
+      - db
+    #    healthcheck:
+    #      test: ["CMD", "docker-healthcheck-coord"]
+    deploy:
+      mode: global
+  db_data_1:
+    environment:
+      - PG_GTM_HOST=db_gtm_1
+      - PG_GTM_PORT=6666
+      - PG_HOST=0.0.0.0
+      - PG_NODE=data_1
+      - PG_PORT=5432
+    image: pavouk0/postgres-xl:latest
+    command: docker-cmd-data
+    entrypoint: docker-entrypoint-data
+    depends_on:
+      - db_gtm_1
+    volumes:
+      - db_data_1:/var/lib/postgresql
+    networks:
+      - db
+    #    healthcheck:
+    #      test: ["CMD", "docker-healthcheck-data"]
+    deploy:
+      mode: global
+  db_data_2:
+    environment:
+      - PG_GTM_HOST=db_gtm_1
+      - PG_GTM_PORT=6666
+      - PG_HOST=0.0.0.0
+      - PG_NODE=data_2
+      - PG_PORT=5432
+    image: pavouk0/postgres-xl:latest
+    command: docker-cmd-data
+    entrypoint: docker-entrypoint-data
+    depends_on:
+      - db_gtm_1
+    volumes:
+      - db_data_2:/var/lib/postgresql
+    networks:
+      - db
+    #    healthcheck:
+    #      test: ["CMD", "docker-healthcheck-data"]
+    deploy:
+      mode: global
+volumes:
+  db_gtm_1: {}
+  db_coord_1: {}
+  db_coord_2: {}
+  db_data_1: {}
+  db_data_2: {}
+networks:
+  db:
```

Comment:
This creates a huge security risk. Since the Postgres-XL cluster doesn't support authentication for inter-node communictions, it needs to use the `trust` in `pg_hba.conf`. But using the same network for both cluster communication and access into the cluster from outside means that any roles and passwords you set will be silently ignored, and anything will be allowed direct access from anywhere. Not only that, but this create a cluster-corruption risk, since non-Postgres-XL services could talk directly to the Datanodes, rather than being forced to go through the Coordinators. This in turn means that Postgres-XL would not maintain its metadata, and the cluster would be highly like to corrupt.

Reply:
OK I didn't realise this. That is a big problem. Thanks for the clarification.